crypto/http/http_lib.c

```diff
@@ -92,8 +92,8 @@ int OSSL_parse_url(const char *url, char **pscheme, char **puser, char **phost,
 
     /* parse optional "userinfo@" */
     user = user_end = host = p;
-    host = strchr(p, '@');
-    if (host != NULL)
+    host = strpbrk(p, "@[]/?#"); /* [ and ] delimit IPv6 addresses */
+    if (host != NULL && *host == '@')
```

Comment:
I should not that the parsing of `<scheme>://` a few lines above is incorrect.  It misinterprets:
```
host.example.org/some/path#://not-a-scheme/not.a.host:404
```
The scheme name has a [restricted syntax](https://datatracker.ietf.org/doc/html/rfc3986#section-3.1) that SHOULD be checked, at least to not contain ':/@#?', but simpler to check the allowed charset.